zio-http/src/main/scala/zio/http/Response.scala

```diff
@@ -14,13 +14,15 @@ import zio.{Cause, Promise, Task, Trace, Unsafe, ZIO}
 
 import java.io.IOException
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
+import java.util.concurrent.atomic.AtomicReference
 
-final case class Response private (
+sealed abstract case class Response private (
   status: Status,
   headers: Headers,
   body: Body,
   private[zio] val attribute: Response.Attribute,
   private[zio] val httpError: Option[HttpError],
+  frozen: Boolean = false,
```

Comment:
Note that we almost never use this `Boolean` field so it may make sense to have `def frozen = false` in the trait, and override for more specific classes.

Reply:
Makes sense.  I can put that on the "TODO" list for the next PR.

Reply:
@jdegoes I went with removing the `frozen` field all together and replaced it with a simple, empty marker trait, `Frozen`.

Reply:
I am rethinking this approach now that I started refactoring `Response` a bit more.

Reply:
'Response` now completely refactored and should address these issues.